### PR TITLE
update Circleci to v2

### LINKED ISSUE #36: coverage is broken
Switching to tox for multi-python testing has broken pytest-coverage

```
Coverage.py warning: No data was collected.
```

### PR DESCRIPTION
**Pull request by vincentsarago**
closes : #36 

cc @perrygeo

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        name: install dependencies
+        command: pip install --user tox codecov
+    - run:
+        name: run tox
+        command: ~/.local/bin/tox
+    - run:
+        name: upload coverage report
+        command: |
+          if [[ "$UPLOAD_COVERAGE" == 1 ]]; then
+             ~/.local/bin/coverage xml
```
Comment (vincentsarago) [inline .circleci/config.yml]: because we use too to set env and run test, we need to run `coverage` outside tox

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - run:
+        name: install dependencies
+        command: pip install --user tox codecov
+    - run:
+        name: run tox
+        command: ~/.local/bin/tox
+    - run:
+        name: upload coverage report
+        command: |
+          if [[ "$UPLOAD_COVERAGE" == 1 ]]; then
```
Comment (vincentsarago) [inline .circleci/config.yml]: `UPLOAD_COVERAGE` variable is set so we don't create coverage for any env (py2 and py3)

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+             ~/.local/bin/coverage xml
+             ~/.local/bin/codecov
```
Comment (vincentsarago) [inline .circleci/config.yml]: We don't need token here because it's a public repo

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+version: 2
+
+common: &common
+  working_directory: ~/raster-tester
+  steps:
+    - checkout
+    - run: sudo apt-get -y --no-install-recommends --force-yes install gdal-bin libgdal-dev
```
Comment (perrygeo) [inline .circleci/config.yml:7]: If we're using Rasterio wheels, we could get away with not installing GDAL, right?

**Review by perrygeo — APPROVED** [review]
👍

File: .circleci/config.yml (hunk 4 above)
Comment (vincentsarago) [reply .circleci/config.yml:7]: yes

**Pull request merged**